app/src/main/java/org/mozilla/fenix/components/metrics/Event.kt

```diff
@@ -319,11 +319,20 @@ sealed class Event {
             get() = hashMapOf(Events.appOpenedAllStartupKeys.source to source.name)
     }
 
-    data class AppOpenedAllSourceStartup(val source: Source) : Event() {
-        enum class Source { APP_ICON, LINK, CUSTOM_TAB, UNKNOWN }
+    data class AppStartupSourceAndType(
+        val source: Source,
+        val type: Type,
+        val hasSavedInstance: Boolean
+    ) : Event() {
+        enum class Source { APP_ICON, LINK, CUSTOM_TAB, ERROR }
+        enum class Type { COLD, WARM, HOT, ERROR }
 
         override val extras: Map<Events.appOpenedAllStartupKeys, String>?
-            get() = hashMapOf(Events.appOpenedAllStartupKeys.source to source.name)
+            get() = hashMapOf(
+                Events.appOpenedAllStartupKeys.source to source.toString(),
+                Events.appOpenedAllStartupKeys.startupType to type.toString(),
+                Events.appOpenedAllStartupKeys.startupType to hasSavedInstance.toString()
```

Comment:
This is the wrong key

Side notes:
- if glean had extras type safety, it would have caught this
- a test for the conversion function would have caught this
- you may not have done this but there's a (tbh, hard to believe) phenomenon that happens where if you copy, paste, and edit the paste, you're more likely to make an error in the last replacement of your copy paste: it's an "off-by-one" problem :P It's a reason to 1) write code by hand instead of copy-pasting when feasible or 2) ensuring code where copy-and-pasting is unnecessary. However, I don't see a clear way to do that here.

Reply:
Ah Idk if I can blame this mistake on that specific phenomenon 😓  but I have definitely come across it while working with xml files